tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/PortalServerBehavior.java

```diff
@@ -210,6 +210,18 @@ public void launchServer(ILaunch launch, String mode, IProgressMonitor monitor)
 				url += ":" + port;
 			}
 
+			IPath appServerPortalPath = _getPortalRuntime().getAppServerPortalDir();
+
+			if (appServerPortalPath != null) {
+				String appServerPortalDir = appServerPortalPath.toString();
+
+				String folderName = appServerPortalDir.substring(appServerPortalDir.lastIndexOf("/"));
```

Comment:
Is the context name always equals to the directory name?  in tomcat how do you specify the custom context name?  Through a directoyr name or through a context.xml file?

Reply:
Hi Greg, in tomcat we can specify custom context name by both of them.

Actually, when deploy applications, Tomcat will always process XML descriptors first, then packaged WARs, and then exploded WARs. And how to get context name from them, that is the work that "PortalContext" class will do. It called "ContextName" in tomcat, it creates an instance from a context name, display name, base name, directory name, War name or context.xml name. Then we can get context name or context path by get method.